src-ocaml/baijiu_blake2b.ml

```diff
@@ -362,38 +356,43 @@ module Make (B : Baijiu_buffer.S)
       ; buf
       ; h = Array.make 8 0L
       ; t = Array.make 2 0L
-      ; f = Array.make 2 0L }
-    in
+      ; f = Array.make 2 0L } in
 
     let param_bytes = param_to_bytes
         { default_param with digest_length = outlen
                            ; key_length = len } in
 
     for i = 0 to 7
-    do ctx.h.(i) <- Int64.(iv.(i) lxor (B.le64_to_cpu param_bytes (i * 8))) done;
+    do ctx.h.(i) <- Int64.(iv.(i) lxor (By.le64_to_cpu param_bytes (i * 8))) done;
 
     if len > 0
     then begin
-      let block = B.create 128 in
+      let block = By.create 128 in
```

Comment:
Seems like `Bytes.make` could also be used here?

Not sure what `blit` accepts in which arguments, but I took it to mean "Put the key into this new block at offset 0" in which case I guess we don't have to set all of them to 0, just the diff `128-len`.

Another thought: what happens if `len > 128`?